simulator/virtual_machine.go

```diff
@@ -1621,8 +1621,9 @@ func (vm *VirtualMachine) CloneVMTask(ctx *Context, req *types.CloneVM_Task) soa
 			return nil, &types.InvalidArgument{InvalidProperty: "spec.location.pool"}
 		}
 		config := types.VirtualMachineConfigSpec{
-			Name:    req.Name,
-			GuestId: vm.Config.GuestId,
+			Name:         req.Name,
+			GuestId:      vm.Config.GuestId,
+			InstanceUuid: req.Spec.Config.InstanceUuid,
```

Comment:
Thanks @parveenchahal 
Just need to move the setting of this field inside the `if req.Spec.Config != nil {` block below to avoid possible panic.
CI failed due to this: https://travis-ci.org/github/vmware/govmomi/jobs/706282464

Reply:
Thanks you, @dougm . I've updated the PR with the change.